[DOCS-14397] Add Bits Data Preview docs page (#36673)
DataDog/documentation: content/en/bits_ai/bits_data/_index.md

---
title: Bits Data
description: "Ask questions about your business data in natural language. Bits Data writes the queries, runs them against your data warehouse, and returns answers without SQL or dashboard hunting."
further_reading:
- link: "/notebooks/"
  tag: "Documentation"
  text: "Build analyses in Datadog Notebooks"
- link: "/dashboards/"
  tag: "Documentation"
  text: "Datadog Dashboards"
- link: "/ddsql_editor/"
  tag: "Documentation"
  text: "Query data with DDSQL"
cascade:
    site_support_id: bits_data
---

{{< callout url="http://datadoghq.com/product-preview/bits-data" >}}
Bits Data is in Preview. Click <strong>Request Access</strong> to join the Preview program.
{{< /callout >}}

## Overview

Bits Data is an agentic AI tool that lets you explore and analyze your business data through natural language. Ask questions about your revenue, customers, support, or sales pipeline. Bits Data identifies the relevant tables, writes the SQL, runs it against your connected data warehouse, and returns the results. Combine business data with your Datadog observability data in dashboards, notebooks, and [DDSQL][1] queries.

> ## Use cases

Reviewer: Few things worth including
* Visualizing results and saving to dashboards

- Ask ad-hoc business questions without writing SQL.
- Combine business and observability data in a single analysis.
- Save reusable [skills](#skills) for repeated workflows.
- Pull results directly into a [Datadog Notebook][2] for further analysis.

## Sample questions

Ask Bits Data questions like:

- `What is the ARR for top customers this quarter?`
- `Which products have the most active customers?`
- `What is the monthly support ticket volume trend?`
- `Which products have the highest open ticket count?`
- `Which deals have been in negotiation longer than 30 days?`
- `What is the company headcount by office location?`

## How it works

1. Type a question in natural language.
2. Bits Data identifies the relevant tables, writes the SQL, and runs it against your connected data warehouse.
3. Results are returned as a table you can explore.
4. Save the results to a notebook, refine the query, or turn it into a skill.

> ## Query contexts

Reviewer: Should we have a separate section for how data teams can define evals against the query context to ensure agent answers business questions correctly? We should at least include it in the query contexts section, it's probably the most important selling point for the product

> Query contexts are curated instructions per data domain, such as Product or Sales. Each context tells Bits Data which tables to use, which filters to apply, and which edge cases to avoid. Query contexts are what make Bits Data accurate and consistent. Data teams maintain query contexts so analysts get reliable answers across the team.

Reviewer: It's important to include that query contexts are auto generated from usage and metadata in existing BI tools + warehouses connected via data observability. 

Also, it is worth including an example here eg. Revenue query context and the types of instructions it will have - the name query contexts is not sufficient IMO

> ## Skills

Reviewer: Also worth including that skills can be shared within your team or even across the company?

---
title: Bits for Data

> description: "Ask questions about your business data in natural language. Bits for Data writes the queries, runs them against your data warehouse, and returns answers without SQL or dashboard hunting."

Reviewer 2: ```suggestion
description: "Ask questions about your business data in natural language. Bits for Data writes the queries, runs them against your data warehouse, and returns answers without writing SQL or navigating dashboards."
```

further_reading:
- link: "/notebooks/"
  tag: "Documentation"
  text: "Build analyses in Datadog Notebooks"
- link: "/dashboards/"
  tag: "Documentation"
  text: "Datadog Dashboards"
- link: "/ddsql_editor/"
  tag: "Documentation"
  text: "Query data with DDSQL"
cascade:
    site_support_id: bits_data
---

> <div class="alert alert-info">Bits for Data is in Preview. Contact your Datadog representative to sign up.</div>

Reviewer 2: just noting this uses the HTML div instead of the callout shortcode

Author: ohh good catch!!

Author: Hi @rtrieu actually I think [this](https://datadoghq.atlassian.net/wiki/spaces/docs4docs/pages/1182958054/Markdown+Shortcode+Reference#Alert-boxes) is correct ? We are not using the Preview form "shortcode" here as the signup form isn't ready yet, this will likely be added post-DASH :)

Reviewer 2: hi @git-thuerk-done, i meant you can use the preview banner without the sign up button temporarily, e.g.

{{< callout url="#" btn_hidden="true" header="false">}}
  Bits for Data is in Preview. Contact your Datadog representative to sign up.
{{< /callout >}}

- Save reusable [skills](#skills) for repeated workflows.
- Pull results directly into a [Datadog Notebook][2] for further analysis.
- Visualize results and save them to a [Datadog Dashboard][3] for ongoing monitoring.

## Sample questions

Ask Bits AI Data questions like:

- `What is the ARR for top customers this quarter?`
- `Which products have the most active customers?`
- `What is the monthly support ticket volume trend?`
- `Which products have the highest open ticket count?`
- `Which deals have been in negotiation longer than 30 days?`
- `What is the company headcount by office location?`

## How it works

1. Type a question in natural language.
2. Bits AI Data identifies the relevant tables, writes the SQL, and runs it against your connected data warehouse.
3. Results are returned as a table you can explore.
4. Save the results to a notebook, refine the query, or turn it into a skill.

## Query contexts

Query contexts are curated instructions per data domain, such as Product or Sales. Each context tells Bits AI Data which tables to use, which filters to apply, and which edge cases to avoid. Query contexts are what make Bits AI Data accurate and consistent.

> Bits AI Data can auto-generate query contexts from usage and metadata in your connected BI tools and data warehouses through [Data Observability][4]. Data teams then review and refine the generated instructions so anyone on the team can get reliable answers, without needing to be an analyst or wait on one.

Reviewer 2: ```suggestion
Bits AI Data can auto-generate query contexts from usage and metadata in your connected BI tools and data warehouses through [Data Observability][4]. Data teams then review and refine the generated instructions so anyone on the team can get reliable answers, without analyst expertise or analyst involvement.
```

### Example: revenue query context

A Revenue query context might include instructions such as:

- Use the `fct_subscription_revenue` table as the primary source for ARR and MRR.
- Filter out internal test accounts (`account_type = 'internal'`) by default.
- Treat revenue as recognized in the currency of the parent account, converted to USD using the `fx_rate_daily` table.
- Exclude churned customers from "active customer" counts unless the question explicitly asks about churn.

> ### Evals

Reviewer 2: ```suggestion
### Evaluations
```
